src/dsp_tools/utils/xmlupload/resource_create_client.py

```diff
@@ -0,0 +1,379 @@
+            case "date":
+                res = _make_date_value(value)
+            case "decimal":
+                res = _make_decimal_value(value)
+            case "geometry":
+                res = _make_geometry_value(value)
+            case "geoname":
+                res = _make_geoname_value(value)
+            case "integer":
+                res = _make_integer_value(value)
+            case "interval":
+                res = _make_interval_value(value)
+            case "resptr":
+                res = _make_link_value(value, self.id2iri_mapping)
+            case "list":
+                res = _make_list_value(value, self.listnode_lookup)
+            case "text":
+                res = _make_text_value(value, self.id2iri_mapping)
+            case "time":
+                res = _make_time_value(value)
+            case "uri":
+                res = _make_uri_value(value)
+            case _:
+                raise UserError(f"Unknown value type: {value_type}")
+        if value.comment:
+            res["knora-api:valueHasComment"] = value.comment
+        if value.permissions:
+            res["knora-api:hasPermissions"] = str(self.permissions_lookup[value.permissions])
+        return res
+
+
+def _make_bitstream_file_value(bitstream_info: BitstreamInfo) -> dict[str, Any]:
+    local_file = Path(bitstream_info.local_file)
+    file_ending = "".join(local_file.suffixes)[1:]
+    match file_ending:
+        case "zip" | "tar" | "gz" | "z" | "tar.gz" | "tgz" | "gzip" | "7z":
+            prop = "knora-api:hasArchiveFileValue"
+            value_type = "ArchiveFileValue"
+        case "mp3" | "wav":
+            prop = "knora-api:hasAudioFileValue"
+            value_type = "AudioFileValue"
+        case "pdf" | "doc" | "docx" | "xls" | "xlsx" | "ppt" | "pptx":
+            prop = "knora-api:hasDocumentFileValue"
+            value_type = "DocumentFileValue"
+        case "mp4":
+            prop = "knora-api:hasMovingImageFileValue"
+            value_type = "MovingImageFileValue"
+        case "jpg" | "jpeg" | "jp2" | "png" | "tif" | "tiff":
+            prop = "knora-api:hasStillImageFileValue"
+            value_type = "StillImageFileValue"
+        case "odd" | "rng" | "txt" | "xml" | "xsd" | "xsl" | "xslt" | "csv":
+            prop = "knora-api:hasTextFileValue"
+            value_type = "TextFileValue"
+        case _:
+            raise UserError(f"Unknown file ending: {file_ending} for file {local_file}")
+    file_value = {
+        "@type": f"knora-api:{value_type}",
+        "knora-api:fileValueHasFilename": bitstream_info.internal_file_name,
+    }
+    if bitstream_info.permissions:
+        file_value["knora-api:hasPermissions"] = str(bitstream_info.permissions)
+    return {prop: file_value}
+
+
+def _make_boolean_value(value: XMLValue) -> dict[str, Any]:
+    return {
+        "@type": "knora-api:BooleanValue",
+        "knora-api:booleanValueAsBoolean": bool(value.value),
```

Comment:
darn... well spotted. Sometimes I wish Python was a more sensible programming language ^^  
How would you do it? ` "knora-api:booleanValueAsBoolean": value.value.lower() in ["true", "1"]`? Or is there a better way?

We definitely need a bool here, because JSON-LD doesn't consider `"true"` the same as `true` and I'm sure the API would be picky about this.

Reply:
Best would be to write our own validation function, something along these lines:
```
match val:
    case "True" | "true" | "1" | 1 | True:
        return True
    case "False" | "false" | "0" | 0 | False:
        return False
```
--> Could you then also update the docs, that uppercase True/False are allowed?